docs/rules/no-required-schema-properties-undefined.md

```diff
@@ -0,0 +1,89 @@
+---
+slug: /docs/cli/rules/no-required-schema-properties-undefined
+---
+
+# no-required-schema-properties-undefined
+
+Ensures there are no required schema properties that are undefined.
+
+| OAS | Compatibility |
+| --- | ------------- |
+| 2.0 | ✅            |
+| 3.0 | ✅            |
+| 3.1 | ✅            |
+
+```mermaid
+flowchart TD
+
+Root ==> Components ==> Schemas
+
+style Schemas fill:#codaf9,stroke:#0044d4,stroke-width:5px
+```
+
+## API design principles
+
+If a required schema property is declared but not defined, this rule informs you which of the required schema properties are missing.
+
+## Configuration
+
+| Option   | Type   | Description                                                                              |
+| -------- | ------ | ---------------------------------------------------------------------------------------- |
+| severity | string | Possible values: `off`, `warn`, `error`. Default `off` (in `recommended` configuration). |
+
+An example configuration:
+
+```yaml
+rules:
+  no-required-schema-properties-undefined: error
+```
+
+## Examples
+
+Given this configuration:
+
+```yaml
+rules:
+  no-required-schema-properties-undefined: error
+```
+
+Example of **incorrect** schema properties:
```

Comment:
Can we see an example of what the linter returns in this case? I think users will search for the error message so it is good to include it in the docs. (Everything else looks good)

Reply:
Fixed, thanks!